pkg/cvo/upgradeable.go

```diff
@@ -152,6 +153,31 @@ func (check *clusterOperatorsUpgradeable) Check() *configv1.ClusterOperatorStatu
 		return cond
 	}
 
+	// Set upgradeable = false to stop the y stream upgrades to happen when an upgrade is in progress.
+	// This will only impact the Y stream updates as CVO ignores this for z stream updates
+	var isUpgradeInProgress bool
+	for _, op := range ops {
+		if sc := resourcemerge.FindOperatorStatusCondition(op.Status.Conditions, configv1.OperatorProgressing); sc != nil {
+			isUpgradeInProgress = true
+		}
+	}
+	now := metav1.Now()
+	if isUpgradeInProgress {
+		var statusConds []configv1.ClusterOperatorStatusCondition
+		statusConds = append(statusConds, configv1.ClusterOperatorStatusCondition{
+			Type:               configv1.OperatorUpgradeable,
+			Status:             configv1.ConditionFalse,
+			Reason:             "UpgradeInProgress",
+			Message:            fmt.Sprintf("Cluster cannot be upgraded between minor versions"),
+			LastTransitionTime: now,
+		})
+		check.operator.setUpgradeable(&upgradeable{
```

Comment:
This really isn’t safe.  Methods called check are supposed to be side effect free.  I think you’re going to have to figure out how to keep the information flow strictly one way here (the sync loop needs to be the one setting this).  Remember, setting spec is intent and so there is no guarantee of an atomic transition from the upgrading step.  And I’m a bit worried that the logic really needs to be part of the state reconciler (from spec to the sync loop) and that is a much bigger change.